addons/event_sale/models/event_registration.py

```diff
@@ -24,20 +24,28 @@ class EventRegistration(models.Model):
     utm_medium_id = fields.Many2one(compute='_compute_utm_medium_id', readonly=False,
         store=True, ondelete="set null")
 
-    @api.depends('sale_order_id.state', 'sale_order_id.currency_id', 'sale_order_line_id.price_total')
+    @api.depends('sale_order_id.state', 'sale_order_id.currency_id', 'sale_order_line_id.price_total', 'sale_order_line_id.product_uom_qty')
     def _compute_registration_status(self):
         self.filtered(lambda reg: not reg.state).state = 'draft'
         for so_line, registrations in self.grouped('sale_order_line_id').items():
             cancelled_registrations = registrations.filtered(lambda reg: reg.sale_order_id.state == 'cancel')
             cancelled_registrations.state = 'cancel'
             if not so_line or float_is_zero(so_line.price_total, precision_digits=so_line.currency_id.rounding):
                 registrations.sale_status = 'free'
-                registrations.filtered(lambda reg: reg.state == 'draft').write({"state": "open"})
+                new_free_registrations = registrations.filtered(lambda reg: reg.state == 'draft')
+                if so_line:
+                    new_free_registrations[:int(so_line.product_uom_qty - len(registrations - new_free_registrations))].write({"state": "open"})
+                else:
+                    new_free_registrations.write({"state": "open"})
```

Comment:
Seems a bit odd to call "write", can't we simply assign the value like it's done here under for "draft" ?